build/source/engine/derivforce.f90

```diff
@@ -241,7 +261,28 @@ subroutine derivforce(err,message)
 
  ! compute density of new snow
  if(snowfall > tiny(fracrain))then
-  newSnowDensity = newSnowDenMin + newSnowDenMult*exp((airtemp-Tfreeze)/newSnowDenScal)  ! new snow density (kg m-3)
+  ! Determine which method to use 
+  select case(model_decisions(iLookDECISIONS%snowDenNew)%iDecision)
+   ! Hedstrom and Pomeroy 1998
+   case(hedAndPom) 
+    newSnowDensity = min(150._dp,newSnowDenMin + newSnowDenMult*exp((airtemp-Tfreeze)/newSnowDenScal))  ! new snow density (kg m-3)
```

Comment:
We'd like to avoid hard-coded constants and parameters in equations. If they are physical constants, they are already defined in `dshare/multiconst.f90` and you should reuse those constant declarations, so we do not have multiple versions of the same constants floating around (for example multiple different names and values for something like ice density. If they are parameters such as the value `150.`, then they should ideally be treated as an input parameter. At the very least they should be declared as a variable at the top of a routine with their own comment, so we know what they represent and there are no hard-coded numbers in the equation.

In this case it is unclear to me whether `150._dp` is really a lower boundary on `newSnowDenMin` and therefore something that should be checked when `newSnowDenMin` is read from file (and not checked as part of the equation) or whether `150._dp` represents a separate parameter.